### PR TITLE
Create CodeQL pipeline

### PR DESCRIPTION
**Pull request by szlatkow**

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v1
+        with:
+          queries: security-and-quality
+
+      - name: Restore solution
+        shell: pwsh
```
Comment (aleks-ivanov) [inline .github/workflows/codeql-analysis.yml]: `windows-latest`, `shell: pwsh` declaration is unnecessary

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        run: nuget restore PerfView.sln
+
+      - name: Build solution
+        shell: pwsh
```
Comment (aleks-ivanov) [inline .github/workflows/codeql-analysis.yml]: `windows-latest`, `shell: pwsh` declaration is unnecessary

File: .github/workflows/codeql-analysis.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  analyze:
+    name: Analyze
+    runs-on: windows-latest
+    steps:
+      - name: Setup .NET SDK
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 2.1.505
+
+      - name: Add MSBuild to PATH
```
Comment (aleks-ivanov) [inline .github/workflows/codeql-analysis.yml]: Does `dotnet` CLI not work even under `windows-latest` ?

**Review by aleks-ivanov — CHANGES_REQUESTED** [review]
Checkout the review comments

File: .github/workflows/codeql-analysis.yml (hunk 3 above)
Comment (szlatkow) [reply .github/workflows/codeql-analysis.yml]: Yes, we established that together

Comment (aleks-ivanov) [reply .github/workflows/codeql-analysis.yml]: @szlatkow alright, add an explanation about that to the PR in the original repo. It is odd that the `dotnet` CLI doesn't work with the project on Windows and the owners should know about it. I am ready to merge this PR whenever you are done making changes to it.

**Review by aleks-ivanov — CHANGES_REQUESTED** [review]
@szlatkow as a last improvement, I would like you not to disregard the CodeQL template that drastically. Even tho you need nuget/msbuild CLI, you can still use the established patterns.

Try to adapt the nuget/msbuild CLI commands to the template, not the other way around.

**Review by aleks-ivanov — APPROVED** [review]

**Pull request merged**